src/OrderManager/V7/OrderManager.php

```diff
@@ -307,10 +310,21 @@ public function recreateOrder(CartInterface $cart): AbstractOrder
             $order->setOrderdate(new Carbon());
             $order->setCartId($sourceOrder->getCartId());
 
+            if ($tokens) {
```

Comment:
move the logic to generic method and consider it also for `recreateOrderBasedOnSourceOrder`

Reply:
didn't find any usage of it, but looks like the funcitonality there is to "copy" an order, looks it doesn't need to copy the tokens over or so. It's unlinked from the cart.